Client/Ecosia/UI/WhatsNew/DataProvider/WhatsNewLocalDataProvider.swift

```diff
@@ -88,21 +93,15 @@ final class WhatsNewLocalDataProvider: WhatsNewDataProvider {
 
         // Gather all versions
         let allVersions = Array(whatsNewItems.keys).sorted()
-
-        // Gather first item in `allVersions` array
-        guard let firstItemInAllVersions = allVersions.first else { return [] }
         
-        // Ensure the `toVersion` is greater than or equal to the smallest version in `whatsNewItems`
-        guard toVersion >= firstItemInAllVersions else { return [] }
-
-        // Find the closest previous version or use the first one if `from` is older than all versions.
-        let fromIndex = allVersions.lastIndex { $0 <= fromVersion } ?? 0
-
-        // Find the index of `to` version or the last version if `to` is newer than all versions.
-        let toIndex = allVersions.firstIndex { $0 >= toVersion } ?? (allVersions.count - 1)
+        // Find the index of the version immediately after `fromVersion`
+        guard let fromIndex = allVersions.firstIndex(where: { $0 > fromVersion }) else { return [] }
+        
+        // Find the index of the version immediately before or equal to `toVersion`
+        guard let toIndex = allVersions.lastIndex(where: { $0 <= toVersion }) else { return [] }
         
-        // Return the range.
-        return Array(allVersions[fromIndex...toIndex])
+        // Return the range between `fromIndex` (excluded) and `toIndex` (included)
+        return Array(allVersions[fromIndex..<toIndex + 1])
```

Comment:
What will happen if `toIndex` is the last index? Will `toIndex + 1` return an error or does the range handle that?

Reply:
Thanks for asking @lucaschifino !
I believe this test `testUpgradeToGreaterVersionThanAnyInBetweenWithItemsShouldShowWhatsNew()` covers your precise case.